packages/profiler-app/src/bundleProcessor.ts

```diff
@@ -0,0 +1,177 @@
+import * as zlib from 'zlib';
+import { promisify } from 'util';
+import AdmZip from 'adm-zip';
+
+const gunzip = promisify(zlib.gunzip);
+
+export interface BundleProcessOptions {
+    bundlePath: string;
+    verbose?: boolean;
+    outputName?: string;
+}
+
+/**
+ * Process a support bundle file:
+ * - Extract profile and config files
+ * - Generate dataflow graph from SQL
+ * - Save files to data/ directory
+ */
+export class BundleProcessor {
+    private verbose: boolean;
+    private outputName: string;
+
+    constructor(private options: BundleProcessOptions) {
+        this.verbose = options.verbose ?? false;
+        this.outputName = options.outputName ?? 'temp';
+    }
+
+    private log(message: string): void {
+        if (this.verbose) {
+            console.log(message);
+        }
+    }
+
+    private async extractProfileFile(zip: AdmZip): Promise<void> {
+        this.log('Extracting profile file...');
+        const entries = zip.getEntries();
+
+        // Find file matching pattern "*_circuit_profile.json"
+        const profileFiles = entries.filter(e =>
+            e.entryName.endsWith('_circuit_profile.json')
+        );
+
+        if (profileFiles.length === 0) {
+            throw new Error('No json profile file found in the bundle');
+        }
+
+        // Choose the last one in sorted order
+        const profileFile = profileFiles.sort((a, b) =>
+            a.entryName.localeCompare(b.entryName)
+        )[profileFiles.length - 1];
+
+        this.log(`Found profile file: ${profileFile.entryName}`);
+
+        // Extract the JSON file directly (no nested zip anymore)
+        const jsonData = profileFile.getData();
+
+        // Write to data directory
+        const outputPath = path.join('data', `${this.outputName}.json`);
+        await fs.writeFile(outputPath, jsonData);
+
+        this.log(`Extracted profile file to: ${outputPath}`);
+    }
+
+    private async extractAndCompileSQL(zip: AdmZip): Promise<void> {
+        this.log('Extracting config file...');
+        const entries = zip.getEntries();
+
+        // Find files matching "pipeline_config.json.gz"
+        const configFiles = entries.filter(e =>
+            e.entryName.endsWith('pipeline_config.json.gz')
+        );
+
+        if (configFiles.length === 0) {
+            this.log('No config file found, skipping SQL compilation');
+            return;
+        }
+
+        // Choose the last one in sorted order
+        const configFile = configFiles.sort((a, b) =>
+            a.entryName.localeCompare(b.entryName)
+        )[configFiles.length - 1];
+
+        this.log(`Found config file: ${configFile.entryName}`);
+
+        // Extract and decompress the config file
+        const gzippedData = configFile.getData();
+        const jsonData = await gunzip(gzippedData);
+        const config = JSON.parse(jsonData.toString('utf-8'));
+
+        const programCode = config.program_code;
+        if (!programCode) {
```

Comment:
I think we should be able to show something even without sources, so we can change this later to a warning